packages/engine/src/camera/systems/CameraSystem.ts

```diff
@@ -267,12 +273,35 @@ function updateSpectator(cameraEntity: Entity) {
   const spectator = getComponent(cameraEntity, SpectatorComponent)
 
   const networkCameraEntity = world.getOwnedNetworkObjectWithComponent(spectator.userId, NetworkCameraComponent)
+
+  if (!networkCameraEntity) {
+    // Go back to normal mode
+    removeComponent(cameraEntity, SpectatorComponent)
+    if (isClient) joinCurrentWorld()
+    return
+  }
+
   const networkTransform = getComponent(networkCameraEntity, TransformComponent)
+  if (!networkTransform) return
+
+  const cameraTransform = getComponent(cameraEntity, TransformComponent)
+  cameraTransform.position.copy(networkTransform.position)
+  cameraTransform.rotation.copy(networkTransform.rotation)
 
-  if (networkTransform) {
-    const cameraTransform = getComponent(cameraEntity, TransformComponent)
-    cameraTransform.position.copy(networkTransform.position)
-    cameraTransform.rotation.copy(networkTransform.rotation)
+  const networkAvatarEntity = world.getUserAvatarEntity(spectator.userId)
+
+  if (!spectator.headDecapReady) {
```

Comment:
I'm not a fan of all this avatar-related logic being brought into the CameraSystem; can't we do this in the AvatarSystem w/ a AvatarHeadDecapComponent?